### ISSUE TITLE
Pulsar SQL can not query the latest ingested event in a topic , but  Pulsar Consumer/Read always can.   

### ISSUE THREAD

**Issue by frank-dkvan** [apache/pulsar#4976]
**Describe the bug**
Pulsar SQL can not query the latest ingested event in a topic , but  Pulsar Consumer/Read always can.   

the testing Pulsar version is 2.4.0

from the two screenshots below, we can see the latest event is always missed in Palsar SQL. but Pulsar reader client can fetch the latest one correctly.

I am new to Pulsar but very interested in Pulsar. don't know this is a bug, or just a kind interesting feature which needs some hidden config I don't know yet?

**Screenshots**
this is the testing code and screenshot of Pulsar Reader:
![image](https://user-images.githubusercontent.com/17067653/63247250-29e16b80-c297-11e9-95be-b5ec5322a36e.png)
this is the testing query SQL and  screenshot in Pulsar SQL CLI:
![image](https://user-images.githubusercontent.com/17067653/63247438-a116ff80-c297-11e9-8354-534b0fb52ddc.png)

**Comment by frank-dkvan** [comment]
looking at the screenshots above, we can find  Pulsar Reader returns 5 records with the latest ingested event,  but Pulsar SQL only returns 4 records without the latest ingested event.   always missing one latest ingested event from Pulsar SQL while new event comes into Pulsar Topic

**Comment by sijie** [comment]
@frank-dkvan this is a known issue of #3828. because Pulsar doesn't flush LAC so when Presto SQL reads data from bookkeeper bypassing broker, it will miss the latest message because the advanced LAC is not flushed back. 

The fix is waiting for BK 4.10 release.

**Comment by frank-dkvan** [comment]
@sijie got it! thanks

**Comment by frank-dkvan** [comment]
> @frank-dkvan this is a known issue of #3828. because Pulsar doesn't flush LAC so when Presto SQL reads data from bookkeeper bypassing broker, it will miss the latest message because the advanced LAC is not flushed back.
> 
> The fix is waiting for BK 4.10 release.

I tested and found this bug  hasn't solved in latest 2.5.0 snapshot with BK 4.10 release yet.   
will this bug get solved in 2.5.0 official release?

**Comment by sijie** [comment]
@frank-dkvan I think it requires a change at Pulsar side. since we are cutting 2.5.0 next week, I don't think we are able to include this in time. We can do this for 2.5.1 or 2.6.0.

you might be able to work around with 4.10 with a bk client configuraiton. @codelipenghui should be able to help you with this.

**Comment by frank-dkvan** [comment]
thanks @sijie 
Hi @codelipenghui wld you pls help with this: providing more details on bk client configuration at pulsar side...sample code if any would be nice..thanks

**Comment by codelipenghui** [comment]
@frank-dkvan @sijie I have create a new pull request #5822 for expose the config in broker.conf.

**Comment by sijie** [comment]
@frank-dkvan 

I think you can add `bookkeeperExplicitLacIntervalInMills=5` in the JVM system properties when starting the broker.

**Comment by codelipenghui** [comment]
@sijie I think we need to use `explicitLacInterval`, this is the name in Bookkeeper client configuration. The `bookkeeperExplicitLacIntervalInMills` is named in broker side.

**Comment by sijie** [comment]
You are right!

**Comment by frank-dkvan** [comment]
> @frank-dkvan
> 
> I think you can add `bookkeeperExplicitLacIntervalInMills=5` in the JVM system properties when starting the broker.

Hi @codelipenghui   "5" means there would be  5-ms delay before we can see the latest message in Pulsar SQL?  Am I understanding correctly?

**Comment by frank-dkvan** [comment]
> @frank-dkvan @sijie I have create a new pull request #5822 for expose the config in broker.conf.

Thanks @codelipenghui

**Comment by codelipenghui** [comment]
close via #5822
